src/PackageVersions/FallbackVersions.php

```diff
@@ -75,11 +83,14 @@ private static function getVersions(string $composerLockFile) : Generator
     {
         $lockData = json_decode(file_get_contents($composerLockFile), true);
 
-        $lockData['packages-dev'] = $lockData['packages-dev'] ?? [];
+        if (array_key_exists('content-hash', $lockData)) {
```

Comment:
I'd rather suggest having separate methods for when a `.lock` and a `installed.json` are used. Yes, there will be duplication, but that's making things a bit more explicit

Reply:
I decided to put the onus of parsing the package data on the `FallbackVersions::getPackageData()` (formerly `FallbackVersions::getComposerLockPath()`), it now only cares about the final data structure. A nice side effect is now this method doesn't care where the data comes from. It just cares that it's in the right format.

Reply:
Nice!